catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -0,0 +1,302 @@
+
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream)
+            throws IOException, CatalogTransformerException {
+        return transform(inputStream, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream, String id)
+            throws IOException, CatalogTransformerException {
+
+        XstreamPathValueTracker pathValueTracker = null;
+        String xml = IOUtils.toString(inputStream);
+        IOUtils.closeQuietly(inputStream);
+
+        try {
+            pathValueTracker = (XstreamPathValueTracker) xstream.fromXML(xml);
+        } catch (XStreamException e) {
+            throw new CatalogTransformerException(TRANSFORM_EXCEPTION_MSG, e);
+        }
+
+        MetacardImpl metacard = toMetacard(pathValueTracker, id);
+
+        metacard.setAttribute(Metacard.METADATA, xml);
+
+        return metacard;
+    }
+
+    private MetacardImpl toMetacard(final XstreamPathValueTracker pathValueTracker, final String id) {
+        MetacardImpl metacard = new MetacardImpl(GMD_METACARD_TYPE);
+
+        if (StringUtils.isNotEmpty(id)) {
+            metacard.setAttribute(Metacard.ID, id);
+        } else {
+
+            metacard.setId(pathValueTracker.getPathValue(FILE_IDENTIFIER_PATH));
+        }
+        addMetacardDates(pathValueTracker, metacard);
+
+        addMetacardType(pathValueTracker, metacard);
+
+        addMetacardCrs(pathValueTracker, metacard);
+
+        addMetacardTitle(pathValueTracker, metacard);
+
+        addMetacardAbstract(pathValueTracker, metacard);
+
+        addMetacardFormat(pathValueTracker, metacard);
+
+        addMetacardResourceUri(pathValueTracker, metacard);
+
+        addMetacardLocation(pathValueTracker, metacard);
+
+        addMetacardSubject(pathValueTracker, metacard);
+
+        return metacard;
+    }
+
+    private void addMetacardDates(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+
+        String dateStr = pathValueTracker.getPathValue(DATE_STAMP_PATH);
+        if (StringUtils.isNotEmpty(dateStr)) {
+            Date date = CswUnmarshallHelper.convertToDate(dateStr);
+            metacard.setModifiedDate(date);
+            metacard.setCreatedDate(date);
+        }
+
+    }
+
+    private void addMetacardType(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+
+        String type = pathValueTracker.getPathValue(CODE_LIST_VALUE_PATH);
+        if (StringUtils.isNotEmpty(type)) {
+            metacard.setContentTypeName(type);
+            metacard.setAttribute(GmdMetacardType.GMD_TYPE, type);
+        }
+
+    }
+
+    private void addMetacardCrs(final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+        StringBuilder crs = new StringBuilder();
+        crs.append("urn:ogc:def:crs:");
+        crs.append(StringUtils.defaultString(pathValueTracker.getPathValue(CRS_AUTHORITY_PATH)));
```

Comment:
Not according to the CSW ISO profile spec.  Unfortunately the GMD schema is poorly defined and there are little restrictions on these fields.